Generate.py

```diff
@@ -200,9 +200,11 @@ def main(args=None) -> Tuple[argparse.Namespace, int]:
                             except Exception as e:
                                 raise Exception(f"Error setting {k} to {v} for player {player}") from e
 
+                    if not getattr(erargs, "name", None):
+                        setattr(erargs, "name", {})  # if the first slot does not define a name we won't have a name obj to look up
                     if path == args.weights_file_path:  # if name came from the weights file, just use base player name
                         erargs.name[player] = f"Player{player}"
-                    elif not erargs.name[player]:  # if name was not specified, generate it from filename
+                    elif player not in erargs.name:  # if name was not specified, generate it from filename
                         erargs.name[player] = os.path.splitext(os.path.split(path)[-1])[0]
                     erargs.name[player] = handle_name(erargs.name[player], player, name_counter)
```

Comment:
Unsure if you'd want to add this to this PR, but this does technically change the fact that previously you could name yourself `false` (which would get changed to the yaml name) and now you cannot name yourself anything boolean in yaml. Could add this to fix it, but maybe there's a better way?
```suggestion
                    erargs.name[player] = handle_name(str(erargs.name[player]), player, name_counter)
```

Reply:
what is expected / what currently works? if you name yourself "false" it currently ignores the name and runs through this code to get the filename instead?

Reply:
If you name yourself `false` (no quotation marks) currently it ends up actually hitting the yaml name rename line because your name evaluates as, well, False. This is clearly unintended, but the name does work just fine on WebHost, so I guess it converts to a string somewhere. I think it would make sense to do the same here so that `name: false`, `name: on` etc. will gen instead of crashing